app/assets/javascripts/initializers/initializeDateTimeHelpers.js

```diff
@@ -0,0 +1,18 @@
+/* global localizeTimeElements */
+
+'use strict';
+
+function initializeDateHelpers() {
+  // Date without year: Jul 12
+  localizeTimeElements(document.querySelectorAll('time.date-no-year'), {
+    month: 'short',
+    day: '2-digit',
```

Comment:
I think this can sort out the zero-padding issue

```suggestion
    day: 'numeric',
```